File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;
int to_2(int num)
{
    int result = 0;
    for (int i = 1; num>0; i*=10)
    {
        int binary = num%2;
        if (binary == 1)
        {
            result +=1;
        }
        num/=2;
    }
    return result;
}
int solution(int n) {
    int answer = 0;
    int orgin_num = to_2(n);
    for ( int i = 1 ; ; i++)
    {
        if ( orgin_num == to_2(n+i))
        {
            answer = n+i;
            break;
        }
    }
    return answer;
}

File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.py
def number_of_binary_zero(num):
    num = bin(num)[2:]
    return str(num).count("1")
def solution(n):
    answer = 0
    origin_zero_count = number_of_binary_zero(n)
    while(1):
        n+=1
        if(number_of_binary_zero(n) == origin_zero_count):
            break
    return n